File: 06_addPolynomials/6.py
def combine_polynomials(P1, P2, len1, len2):
    size = max(len1, len2)
    result = [0 for _ in range(size)]

    for i in range(len1):
        result[i] = P1[i]

    for i in range(len2):
        result[i] += P2[i]

    return result

def display_polynomial(poly, length):
    for i in range(length):
        print(poly[i], end="")
        if i != 0:
            print(f"x^{i}", end="")
        if i != length - 1:
            print(" + ", end="")
    print()


P1 = [3, 7, 2, 5]
P2 = [6, 1, 8]

len1 = len(P1)
len2 = len(P2)

print("Polynomial A:")
display_polynomial(P1, len1)

print("Polynomial B:")
display_polynomial(P2, len2)

merged = combine_polynomials(P1, P2, len1, len2)
final_len = max(len1, len2)

print("Resultant Polynomial:")
display_polynomial(merged, final_len)

File: 06_addPolynomials/6.c
#include <stdio.h>
#define MAX 100

void combinePolynomials(int P1[], int len1, int P2[], int len2, int result[]) {
    int size = (len1 > len2) ? len1 : len2;

    for (int i = 0; i < size; i++) {
        result[i] = (i < len1) ? P1[i] : 0;
    }

    for (int i = 0; i < len2; i++) {
        result[i] += P2[i];
    }
}

void displayPolynomial(int poly[], int length) {
    for (int i = 0; i < length; i++) {
        printf("%d", poly[i]);
        if (i != 0) {
            printf("x^%d", i);
        }
        if (i != length - 1) {
            printf(" + ");
        }
    }
    printf("\n");
}

int main() {
    int P1[] = {3, 7, 2, 5};      
    int P2[] = {6, 1, 8};    
    int result[MAX];

    int len1 = sizeof(P1) / sizeof(P1[0]);
    int len2 = sizeof(P2) / sizeof(P2[0]);
    int finalLen = (len1 > len2) ? len1 : len2;

    printf("Polynomial A:\n");
    displayPolynomial(P1, len1);

    printf("Polynomial B:\n");
    displayPolynomial(P2, len2);

    combinePolynomials(P1, len1, P2, len2, result);

    printf("Resultant Polynomial:\n");
    displayPolynomial(result, finalLen);

    return 0;
}